### DISCUSSION TITLE
vercel dev serverless functions not working but works in prod

### DISCUSSION THREAD

**Discussion by cantoramann** [vercel/vercel#8312] [Help]
I'm using [Quasar](https://quasar.dev/) with Vercel. When running `vercel.dev` the frontend works well but the serverless functions give 404. When I deploy the project, though, they work very well.

I get the 404 directly from the client side. My api path on localhost is /api, which works when deployed with the exact path.
<img width="1052" alt="Screen Shot 2022-08-03 at 17 28 09" src="https://user-images.githubusercontent.com/46545763/182737821-a0ad5021-be31-4420-91ba-a9c51ddf34f6.png">

My `package.json` script:

```
    "lint": "eslint --ext .js,.ts,.vue ./",
    "format": "prettier --write \"**/*.{js,ts,vue,css,html,md,json}\" --ignore-path .gitignore",
    "test": "echo \"No test specified\" && exit 0",
    "build": "quasar build",
    "deploy": "vercel"
```

Also, my `verce.json` is:

```
{
  "routes": [
    { "handle": "filesystem" },
    { "src": "/.*", "dest": "/index.html" }
  ]
}
```

The output directory in Vercel setings is: `dist/spa`

Does anyone have any idea? If anyone is familiar with Quasar, is there anything I should add to `quasar.config.js`?

**Comment by amyegan** [comment]
Thanks for reaching out! I am not familiar with Quasar, but I can try to help. Do you see any errors in the terminal or in the browser console? 

The [Vercel CLI docs](https://vercel.com/docs/cli#commands/dev/when-to-use-this-command) suggest you might not need to use `vercel dev` if your framework's local build command can already handle everything for you. Possibly a dumb question since I'm not familiar with Quasar: Does it work if you run it using your Quasar's own local dev command?

**Comment by cantoramann** [comment]
Oh no, it does not work in the local build.

**Comment by amyegan** [comment]
This is going to be difficult to solve without seeing the problem for myself. I can take a closer look if you are able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or your repo.

**Comment by cohlar** [comment]
Hi, I'm having the same issue. @cantoramann were you able to figure that out?

@amyegan I've created a repo to demonstrate the issue: https://github.com/cohlar/vercel-quasar-serverless
